pkg/controller/kogitoinfra/mongodb.go

```diff
@@ -117,33 +119,46 @@ func (i *mongoDBInfraReconciler) getMongoDBAppProps(mongoDBInstance *mongodb.Mon
 		if err != nil {
 			return nil, err
 		}
-		appProps[propertiesMongoDBSpring[appPropMongoDBHost]] = mongoDBParsedURL.Hostname()
-		appProps[propertiesMongoDBSpring[appPropMongoDBPort]] = mongoDBParsedURL.Port()
-		appProps[propertiesMongoDBQuarkus[appPropMongoDBURI]] = mongoDBURI
+		if runtime == v1beta1.QuarkusRuntimeType {
+			appProps[propertiesMongoDBQuarkus[appPropMongoDBURI]] = mongoDBURI
+		} else if runtime == v1beta1.SpringBootRuntimeType {
+			appProps[propertiesMongoDBSpring[appPropMongoDBHost]] = mongoDBParsedURL.Hostname()
+			appProps[propertiesMongoDBSpring[appPropMongoDBPort]] = mongoDBParsedURL.Port()
+		}
 	}
 
 	return appProps, nil
 }
 
-func (i *mongoDBInfraReconciler) updateMongoDBAppPropsInStatus(mongoDBInstance *mongodb.MongoDB) error {
-	log.Debug("going to Update MongoDB app properties in kogito infra instance status")
-	appProps, err := i.getMongoDBAppProps(mongoDBInstance)
+func (i *mongoDBInfraReconciler) getMongoDBRuntimeProps(mongoDBInstance *mongodb.MongoDB, runtime v1beta1.RuntimeType) (v1beta1.RuntimeProperties, error) {
+	runtimeProps := v1beta1.RuntimeProperties{}
+	appProps, err := i.getMongoDBRuntimeAppProps(mongoDBInstance, runtime)
 	if err != nil {
-		return err
+		return runtimeProps, err
 	}
-	i.instance.Status.AppProps = appProps
-	log.Debugf("Following app properties are set infra status : %s", appProps)
-	return nil
+	runtimeProps.AppProps = appProps
+
+	envVars, err := i.getMongoDBRuntimeSecretEnvVars(mongoDBInstance, runtime)
+	if err != nil {
+		return runtimeProps, err
+	}
+	runtimeProps.Env = envVars
+
+	return runtimeProps, nil
 }
 
-func (i *mongoDBInfraReconciler) updateMongoDBEnvVarsInStatus(mongoDBInstance *mongodb.MongoDB) error {
-	log.Debug("going to Update MongoDB env properties in kogito infra instance status")
-	envVars, err := i.getMongoDBSecretEnvVars(mongoDBInstance)
+func (i *mongoDBInfraReconciler) updateMongoDBRuntimePropsInStatus(mongoDBInstance *mongodb.MongoDB, runtime v1beta1.RuntimeType) error {
+	log.Debugf("going to Update MongoDB %s runtime vars in kogito infra instance status", runtime)
+	runtimeProps, err := i.getMongoDBRuntimeProps(mongoDBInstance, runtime)
 	if err != nil {
 		return err
 	}
-	i.instance.Status.Env = envVars
-	log.Debugf("Following env properties are set infra status : %s", envVars)
+
+	if i.instance.Status.RuntimeProperties == nil {
+		i.instance.Status.RuntimeProperties = v1beta1.RuntimePropertiesMap{}
+	}
+	i.instance.Status.RuntimeProperties[runtime] = runtimeProps
+	log.Debugf("Following MongoDB %s runtime vars are set infra status : %v", runtime, runtimeProps)
 	return nil
```

Comment:
These functions seem very alike. Can you refactor to one function shared among them?

Reply:
I refactored it to `setRuntimeProperties` in `manage_status.go`.